cpp/src/slice2java/Gen.cpp

```diff
@@ -3049,12 +3049,26 @@ Slice::Gen::TypesVisitor::visitEnum(const EnumPtr& p)
     }
     out << ';';
 
-    out << sp << nl << "public int value()";
+    out << sp;
+    out << nl << "/**";
+    out << nl << " * Returns the integer value of this enumerator.";
+    out << nl << " *";
+    out << nl << " * @return the integer value of this enumerator";
+    out << nl << " */";
+    out << nl << "public int value()";
     out << sb;
     out << nl << "return _value;";
     out << eb;
 
-    out << sp << nl << "public static " << name << " valueOf(int v)";
+    out << sp;
+    out << nl << "/**";
+    out << nl << " * Returns the {@code " << name << "} enumerator corresponding to the given integer value.";
+    out << nl << " *";
+    out << nl << " * @param v the value to match";
+    out << nl << " * @return the {@code " << name
+        << "} enumerator corresponding to the given integer value, or {@code null} if no such type exists";
```

Comment:
Enumerator? The type definitely exists here.
```suggestion
        << "} enumerator corresponding to the given integer value, or {@code null} if no such enumerator exists";
```